app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel_views/promote/promote.jsx

```diff
@@ -1,115 +1,88 @@
+    console.log('dispatch')
+    dispatch(addPromotionalTweetData(this.state));
   }
 
   render() {
-    return (
-      <div className='col-md-9 content-box'>
-        <div className='content-header'>
-          <p className='signal-type-label'> SEND TWEET </p>
-        </div>
-
-        <div className='response-info'>
-          <h4>Promote:</h4>
-          <SignalIcon type='twitter'/>
-          <h4 className='subheading'>@Brand #Offers</h4>
-        </div>
+    const { signal } = this.props;
 
-        <div className='tip-box'>
-          <SignalIcon type='tip'/>
-          <h5>Tip</h5>
-          <p> Increase the awareness of your signal, promote it to your audience </p>
-        </div>
-
-        <div className='promote-box'>
-          <div className='response-text'>
-            <h5>Promotional Tweet</h5>
-            <p>140 Character Limit</p>
+    if (signal.id) {
+      return (
+        <div className='col-md-9 content-box'>
+          <div className='content-header'>
+            <p className='signal-type-label'> SEND TWEET </p>
           </div>
-
-          <div className='input-box'>
-            <FormControl onChange={this.handleChange} componentClass="textarea" placeholder={'Searching for deals any time? Tweet or message #Deals to @Brand'}/>
+          <div className='response-info'>
+            <h4>Promote:</h4>
+            <SignalIcon type='twitter'/>
+            <h4 className='subheading'>@Brand #Offers</h4>
           </div>
-
-          <div className='subheader'>
-            <h5>Promotional Image</h5>
-            <p>Select an image to include or upload your own</p>
+          <div className='tip-box'>
+            <SignalIcon type='tip'/>
+            <h5>Tip</h5>
+            <p> Increase the awareness of your signal, promote it to your audience </p>
           </div>
-
-          <div className='row'>
-            <div className='col-xs-12 col-sm-12 col-md-12 col-lg-12 center promote-image'>
-              { this.showPromoImage() }
+          <div className='promote-box'>
+            <div className='subheader'>
+              <h5>Promotional Tweet</h5>
+              <p>140 Character Limit</p>
+            </div>
+            <div className='promote-input-box'>
+              <FormControl onChange={this.handleChange} componentClass="textarea" placeholder={'Searching for deals any time? Tweet or message #Deals to @Brand'}/>
             </div>
+            <div className='subheader'>
+              <h5>Promotional Image</h5>
+              <p>Select an image to include or upload your own</p>
+            </div>
+            <div className='row'>
+              <div className='col-xs-12 col-sm-12 col-md-12 col-lg-12 center promote-image'>
+                <PromoteImage image={this.state.image} handleImage={this.handleImage} />
+              </div>
+            </div>
+            <Button onClick={this.handleSubmit} type='submit' className='save-btn post-to-timeline-btn'>POST TO YOUR TIMELINE</Button>
```

Comment:
If you change this to not be `type="submit"` it won't try to update the listen signal.